platform-connectors/pkg/nodemetadata/factory.go

```diff
@@ -0,0 +1,91 @@
+// Copyright (c) 2025, NVIDIA CORPORATION.  All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nodemetadata
+
+import (
+	"context"
+	"fmt"
+	"sync"
+)
+
+// ProcessorType defines the type of node metadata processor.
+type ProcessorType string
+
+const (
+	// ProcessorTypeKubernetes creates a processor that fetches metadata from Kubernetes API.
+	ProcessorTypeKubernetes ProcessorType = "kubernetes"
+	// ProcessorTypeSlurm can be added in the future for Slurm-based clusters.
+	// ProcessorTypeSlurm ProcessorType = "slurm"
+)
+
+// ProcessorCreator is a function that creates a processor of a specific type.
+type ProcessorCreator func(ctx context.Context, config *Config, params interface{}) (Processor, error)
```

Comment:
I don't follow this, we know what types of processors we have in code, why can't we call the initialize functions when required? Do we need to make it this generic?